### ISSUE TITLE
Support HTCondor as job queue backend

### ISSUE THREAD

**Issue by mapsacosta** [dask/dask-gateway#440]
We are interested on implementing a job queue backend class for dask-gateway capable of submitting dask clusters to HTCondor pools. I am actively working on this and am also happy to collaborate if this is something other people are also interested in (https://github.com/mapsacosta/dask-gateway/tree/htcondor_backend)

Context:
HTCondor is a specialized workload management system for compute-intensive jobs. Like other full-featured batch systems, HTCondor provides a job queueing mechanism, scheduling policy, priority scheme, resource monitoring, and resource management. Users submit their serial or parallel jobs to HTCondor, HTCondor places them into a queue, chooses when and where to run the jobs based upon a policy, carefully monitors their progress, and ultimately informs the user upon completion. 

https://research.cs.wisc.edu/htcondor/

**Comment by jcrist** [comment]
Hi @mapsacosta! Thanks for opening this, we'd be happy to accept a htcondor backend to dask-gateway. Apologies that custom backends aren't as documented as I'd like, hopefully the pbs and slurm backends provide a decent starting point. Is there anything we can do to help with this effort?

**Comment by consideRatio** [comment]
Thanks for linking out to your work @mapsacosta!!

At this point, I'll go ahead and close this issue as it went stale, but also because I think the situation has changed a bit where I consider is to not have sufficient maintenance capacity to help maintain additional implementations within this repo. I think the pattern to encourage going onwards is that additional backends are to live in a separate repository and maintained by the author.

**Comment by mapsacosta** [comment]
Hi @jcrist @consideRatio we ended up taking a similar approach (delegating as much as we could to an external 'plugin' for Gateway). 

I have implemented a heterogeneous cluster mode combining the Kubernetes backend with HTCondor. When our users create clusters, the server launches a Kubernetes Scheduler pod and when scaling/adapting, the client submits HTCondor jobs to our cluster (~60k cores) and starts Dask workers within Singularity containers. 
These workers join the scheduler and then operate as a regular Dask cluster, which allows us to pick up a LOT of resources from our batch system in a multi-tenant environment. Is this something you guys have considered? Any interest on sharing some thoughts about this?

![image](https://user-images.githubusercontent.com/13898042/164282025-0626b3d3-862e-487a-aa6f-1304e0a85c35.png)
